core/src/repair_generic_traversal.rs

```diff
@@ -125,9 +125,21 @@ pub fn get_closest_completion(
                 let shred_index = blockstore.get_index(slot).unwrap();
                 let dist = if let Some(shred_index) = shred_index {
                     let shred_count = shred_index.data().num_shreds() as u64;
-                    last_index - shred_count
+                    debug_assert!(
+                        last_index.saturating_add(1) >= shred_count,
+                        "last_index={} shred_count={}",
+                        last_index,
+                        shred_count
+                    );
```

Comment:
what do you think of logging a `datapoint_error` here if `last_index.saturating_add(1) < shred_count`? We track the datapoint errors in our metrics so we'll see it

Reply:
changed to datapoint_error